### DISCUSSION TITLE
Unofficial Vite Plugin for LWC

### DISCUSSION THREAD

**Discussion by cardoso** [salesforce/lwc#4800] [Show and tell]
Hi folks!

I'm excited to share a new Vite plugin for LWC that I’ve been working on over the past few weeks. This plugin is aimed at making LWC development faster and easier by leveraging Vite’s speed and simplicity. It's a wrapper over the official rollup plugin with some tweaks to make it work with Vite.

### Setup

Getting started with this Vite plugin is straightforward. Here's a quick example:

```javascript
import { defineConfig } from "vite";
import lwc from "vite-plugin-lwc";

export default defineConfig({
  plugins: [
    lwc(),
  ],
});
```

### Examples

I've created a couple of examples to help you get started:

- **Interactive Example on Stackblitz**: [Stackblitz Example](https://stackblitz.com/~/github.com/cardoso/vite-plugin-lwc-example)
- **GitHub Repository**: [vite-plugin-lwc-example](https://github.com/cardoso/vite-plugin-lwc-example)
- **Live Demo on GitHub Pages**: [GitHub Pages](https://cardo.so/vite-plugin-lwc-example)

### Plugin Links

- **NPM Package**: [vite-plugin-lwc](https://www.npmjs.com/package/vite-plugin-lwc)
- **GitHub Repository**: [vite-plugin-lwc Repository](https://github.com/cardoso/vite-plugin-lwc/tree/main/packages/vite-plugin-lwc)

Feel free to check it out and let me know your thoughts! I’d love to hear feedback, suggestions, or any issues you run into.

**Comment by nolanlawson** [comment]
This is awesome! 🎉 

One thing I wonder about is whether it still makes sense to keep pushing on https://github.com/vitejs/vite/issues/17700 to make `.css` files work better in Vite for use cases like LWC's (or other web component frameworks). The `vite:css` and `vite:css-post` [workarounds](https://github.com/cardoso/vite-plugin-lwc/blob/2430de2453417a34ea32b8ddd29ac8f53b3d4f0e/packages/vite-plugin-lwc/src/index.ts#L8-L13) are great, but it feels like Vite could still be more friendly to web component frameworks (which don't assume that `import 'foo.css'` means "hoist `foo.css` into the global `<head>`", and instead assume that you're either using constructable stylesheets or, in LWC's case, you're using LWC's `static stylesheets` system).
